pkg/controller/daemon/daemon_controller.go

```diff
@@ -723,6 +723,7 @@ func (dsc *DaemonSetsController) updateNode(old, cur interface{}) {
 
 	if reflect.DeepEqual(oldNode.Labels, curNode.Labels) &&
 		reflect.DeepEqual(oldNode.Spec.Taints, curNode.Spec.Taints) &&
+		reflect.DeepEqual(oldNode.Status.Allocatable, curNode.Status.Allocatable) &&
 		nodeInSameCondition(oldNode.Status.Conditions, curNode.Status.Conditions) {
 		// If node labels, taints and condition didn't change, we can ignore this update.
 		return
```

Comment:
Actually, this code block is becoming a pain and much less useful. I'd suggest removing this if-block altogether.

Reply:
Done. PTAL!